backend/sirius-web-emf/src/main/java/org/eclipse/sirius/web/emf/view/ToolInterpreter.java

```diff
@@ -0,0 +1,200 @@
+            if (newValue.asObject().isPresent()) {
+                Object instance = this.ecore.eAdd(optionalSelf.get(), setValueOperation.getFeatureName(), newValue.asObject().get());
+                if (instance != null) {
+                    return this.executeOperations(setValueOperation.getChildren(), variableManager);
+                }
+            }
+        }
+        return Optional.empty();
+    }
+
+    private Optional<VariableManager> executeUnsetValue(VariableManager variableManager, UnsetValue unsetValueOperation) {
+        var optionalSelf = variableManager.get(VariableManager.SELF, EObject.class);
+        if (optionalSelf.isPresent()) {
+            var self = optionalSelf.get();
+            EStructuralFeature feature = self.eClass().getEStructuralFeature(unsetValueOperation.getFeatureName());
+            if (feature != null) {
+                List<EObject> elementsToUnset = this.computeElementsToUnset(variableManager.getVariables(), unsetValueOperation.getElementExpression());
+                this.unset(self, feature, elementsToUnset);
+                return this.executeOperations(unsetValueOperation.getChildren(), variableManager);
+            }
+        }
+        return Optional.empty();
+    }
+
+    private List<EObject> computeElementsToUnset(Map<String, Object> variables, String elementExpression) {
+        List<EObject> elementsToUnset = null;
+        if (elementExpression != null && !elementExpression.isBlank()) {
+            Optional<List<Object>> optionalObjectsToUnset = this.interpreter.evaluateExpression(variables, elementExpression).asObjects();
+            if (optionalObjectsToUnset.isPresent()) {
+                // @formatter:off
+                elementsToUnset = optionalObjectsToUnset.get().stream()
+                                      .filter(EObject.class::isInstance)
+                                      .map(EObject.class::cast)
+                                      .collect(Collectors.toList());
+                // @formatter:on
+            }
+        }
+        return elementsToUnset;
+    }
+
+    private void unset(EObject context, EStructuralFeature featureToEdit, List<EObject> elementsToUnset) {
+        if (elementsToUnset == null) {
+            if (featureToEdit.isMany()) {
+                EList<?> values = (EList<?>) context.eGet(featureToEdit);
+                values.clear();
+            } else {
+                context.eSet(featureToEdit, null);
+            }
+        } else {
+            elementsToUnset.stream().forEach(value -> EcoreUtil.remove(context, featureToEdit, value));
+        }
+    }
+
+    private Optional<VariableManager> executeCreateInstance(VariableManager variableManager, CreateInstance creatInstanceOperation) {
+        var optionalSelf = variableManager.get(VariableManager.SELF, EObject.class);
+        if (optionalSelf.isPresent()) {
+            var newInstance = this.ecore.createInstance(creatInstanceOperation.getTypeName());
```

Comment:
This does not work in the context of a dynamically create EPackage. I have a patch which uses the same workaround as in `CanonicalBehaviors.createSemanticInstance` instead.